hack/test-templates.sh

```diff
@@ -192,7 +192,9 @@ if [ "$got" != "$expected" ]; then
 fi
 
 INFO "Testing limactl copy command"
-tmpfile="$HOME/lima-hostname"
+tmpdir=$(mktemp -d /var/tmp/lima-test-templates.XXXXXX)
+defer "rm -rf \"$tmpdir\""
+tmpfile="$tmpdir/lima-hostname"
 rm -f "$tmpfile"
 limactl cp "$NAME":/etc/hostname "$tmpfile"
 defer "rm -f \"$tmpfile\""
```

Comment:
Do we need to to remove the tempfile now? the temporary directory will be removed anyway.